modules/highgui/src/window_w32.cpp

```diff
@@ -1874,21 +1874,41 @@ static void showSaveDialog(CvWindow* window)
     ofn.lStructSize = sizeof(ofn);
 #endif
     ofn.hwndOwner = window->hwnd;
+#ifdef HAVE_PNG   
     ofn.lpstrFilter = "Portable Network Graphics files (*.png)\0*.png\0"
+#else
+    ofn.lpstrFilter = "Windows bitmap (*.bmp;*.dib)\0*.bmp;*.dib\0"
+#endif
+#ifdef HAVE_JPEG
                       "JPEG files (*.jpeg;*.jpg;*.jpe)\0*.jpeg;*.jpg;*.jpe\0"
+#endif
+#ifdef HAVE_PNG 
                       "Windows bitmap (*.bmp;*.dib)\0*.bmp;*.dib\0"
```

Comment:
I mean that second ".bmp" entry can be removed:
```
    ofn.lpstrFilter = 
#ifdef HAVE_PNG
                      "Portable Network Graphics files (*.png)\0*.png\0"
#endif
                      "Windows bitmap (*.bmp;*.dib)\0*.bmp;*.dib\0"
#ifdef HAVE_JPEG
                      "JPEG files (*.jpeg;*.jpg;*.jpe)\0*.jpeg;*.jpg;*.jpe\0"
#endif
...
```

P.S. But it will change list order.